plugins/hdf/io/HdfReader.cpp

```diff
@@ -102,10 +105,11 @@ point_count_t HdfReader::read(PointViewPtr view, point_count_t count)
 }
 
 
-bool HdfReader::processOne(PointRef& point) {
-    for(int index = 0; index < m_infos.size(); ++index) {
+bool HdfReader::processOne(PointRef& point)
```

Comment:
There are a few things here that can be improved, but I think the big thing to me is that it brings all the HDF logic into this function instead of the handler.  That means that you have to store a vector<void *> in this class and know how things are laid out in the HDF5Handler, which seems unfortunate to me.  I think I'd just want to fetch a pointer to the data for each field in the point from the handler.  Something like:
```
for (DimInfo& di : dimensions)
{
  void *fieldData = m_hdf5Handler.value(di, m_index);
  point.setField(di.id, di.pdal_type, fieldData);
}
```
All the rest can be done in the handler.  That means that this code has to know nothing about chunks or store information about the handler's data.

The other thing I'd be careful of is naming.  Calling lots of thing "index" is confusing, especially when they're different.  `m_index` is a point ID.  `index` is a loop variable.  You can eliminate the loop variable by using what's called a "range-based for loop."  It's a terrible name, but it just lets you get back the actual object in the collection you're looping.  In your case:

```
for (auto& dimInfo : m_infos)
...
```
I don't love `auto` if the type is simple.  If you do:

```
for (DimInfo& dimInfo : m_infos)
  ...
```

I think it's more clear for the reader - you have to do less searching if you're not using an IDE.  And again, naming matters.  Calling something `info` really doesn't convey much information.  That said, sometimes coming up with a good name is hard.  Also, if you explicitly use the type when defining the name and the scope of the name is small, then I don't much mind index-type/shortcut names.  But if you use something like `auto` and an ambiguous name like `info`, then I'm left searching around...  I have to go figure out the type of `m_infos` to know what's going on.